### ISSUE TITLE
Stream cleaners should call to_s on item to be cleaned

### ISSUE THREAD

**Issue by agross** [nanoc/nanoc#775]
Currently it's a bit inconvenient to do printf-debugging in your content or layout files:

```
<% puts @items.class %>
=> NoMethodError: undefined method `gsub' for Nanoc::ItemCollectionWithRepsView:Class
```

Couldn't the stream cleaner just call `#to_s` before it does its cleaning job?

**Comment by ddfreyne** [comment]
This was fixed in #712 and will be part of the upcoming 4.1 release.

**Comment by agross** [comment]
I'm running RC2:

```
Captain! We've been hit!

Message:

NoMethodError: undefined method `gsub' for #<Nanoc::ItemWithRepsView:0x00000002fbe738>

Compilation stack:

  - [item]   /navigation/child-1/index.html (rep default)

Stack trace:

  0. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/stream_cleaners/utf8.rb:9:in `clean'
  1. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:143:in `block in _nanoc_clean'
  2. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:143:in `each'
  3. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:143:in `reduce'
  4. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:143:in `_nanoc_clean'
  5. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:81:in `block (2 levels) in puts'
  6. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:81:in `map'
  7. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:81:in `block in puts'
  8. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:147:in `_nanoc_swallow_broken_pipe_errors_while'
  9. C:/Ruby/lib/ruby/gems/2.2.0/gems/nanoc-4.1.0rc2/lib/nanoc/cli/cleaning_stream.rb:80:in `puts'
  ... 47 more lines omitted. See full crash log for details.
```

Seems lik UTF-8 doesn't call `to_s`.

**Comment by ddfreyne** [comment]
Fixed in #778. Please give it a try and shout if it doesn’t work as expected!
